Android/app/src/main/java/app/intra/MainActivity.java

```diff
@@ -472,6 +480,30 @@ private String getSystemDnsServer() {
     return null;
   }
 
+  private enum PrivateDnsMode { NONE, OPPORTUNISTIC, STRICT };
```

Comment:
I think everything is fine here, but just wanted to raise that the PrivateDnsMode that gets detected by Intra will be the private dns mode "in practice" on the Android OS rather than the private dns mode selected in the Android OS settings.  This discrepancy is due to the case where linkProperties.getPrivateDnsServerName() == null && linkProperties.isPrivateDnsActive(), which means that the Android OS setting is "opportunistic" AND we have a validated resolver.

Reply:
You're right: `OPPORTUNISTIC` is a potentially confusing name for this enum value.  I'll change it to `UPGRADED`, to reflect the idea that the current state indicates a successful auto-upgrade to TLS, and add a comment.